# HG changeset patch
# User Peter Szücs <peter.szucs.dev@gmail.com>
# Date 1406883415 -7200
#      Fri Aug 01 10:56:55 2014 +0200
# Branch v1-9
# Node ID 2b38c1d049506784e1232255bc2f1e70f5c953f1
# Parent  7c776867621e1e41f1be5e49ecc01e8849a80ad2
Fix Ogre for Android Standalone toolchain.

diff -r 7c776867621e -r 2b38c1d04950 CMake/ConfigureBuild.cmake
--- a/CMake/ConfigureBuild.cmake	Fri Jun 20 00:39:55 2014 +0200
+++ b/CMake/ConfigureBuild.cmake	Fri Aug 01 10:56:55 2014 +0200
@@ -183,7 +183,7 @@
 
 
 # Create the pkg-config package files on Unix systems
-if (UNIX)
+if (ANDROID OR UNIX)
   set(OGRE_LIB_SUFFIX "")
   set(OGRE_PLUGIN_PREFIX "")
   set(OGRE_PLUGIN_EXT ".so")
@@ -206,10 +206,12 @@
   endif ()
   if (OGRE_STATIC)
     if (OGRE_CONFIG_THREADS)
-      set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lboost-thread-mt")
+      set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lboost_thread -lboost_system -lboost_atomic")
     endif ()
     # there is no pkgconfig file for freeimage, so we need to add that lib manually
-    set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lfreeimage")
+    set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lFreeImage")
+    # there is no pkgconfig file for freetype, so we need to add that lib manually
+    set(OGRE_ADDITIONAL_LIBS "${OGRE_ADDITIONAL_LIBS} -lfreetype")
     configure_file(${OGRE_TEMPLATES_DIR}/OGREStatic.pc.in ${OGRE_BINARY_DIR}/pkgconfig/OGRE.pc @ONLY)
   else ()
     configure_file(${OGRE_TEMPLATES_DIR}/OGRE.pc.in ${OGRE_BINARY_DIR}/pkgconfig/OGRE.pc @ONLY)
diff -r 7c776867621e -r 2b38c1d04950 CMake/Templates/OGRE.pc.in
--- a/CMake/Templates/OGRE.pc.in	Fri Jun 20 00:39:55 2014 +0200
+++ b/CMake/Templates/OGRE.pc.in	Fri Aug 01 10:56:55 2014 +0200
@@ -8,5 +8,5 @@
 Description: Object-Oriented Graphics Rendering Engine
 Version: @OGRE_VERSION@
 URL: http://www.ogre3d.org
-Libs: -L${libdir} -lOgreMain@OGRE_LIB_SUFFIX@ @OGRE_ADDITIONAL_LIBS@
-Cflags: -I${includedir} -I${includedir}/OGRE @OGRE_CFLAGS@
+Libs: -L${libdir} -L${plugindir} -lPlugin_ParticleFX@OGRE_LIB_SUFFIX@ -lRenderSystem_GLES2@OGRE_LIB_SUFFIX@ -lOgreMain@OGRE_LIB_SUFFIX@ @OGRE_ADDITIONAL_LIBS@ -lboost_thread -lboost_system -lboost_atomic -lFreeImage -lfreetype -lzzip
+Cflags: -I${includedir} -I${includedir}/OGRE @OGRE_CFLAGS@
\ No newline at end of file
diff -r 7c776867621e -r 2b38c1d04950 CMake/Templates/OGREStatic.pc.in
--- a/CMake/Templates/OGREStatic.pc.in	Fri Jun 20 00:39:55 2014 +0200
+++ b/CMake/Templates/OGREStatic.pc.in	Fri Aug 01 10:56:55 2014 +0200
@@ -8,6 +8,6 @@
 Description: Object-Oriented Graphics Rendering Engine
 Version: @OGRE_VERSION@
 URL: http://www.ogre3d.org
-Requires: freetype2, zziplib, x11, xt, xaw7, gl
-Libs: -L${libdir} -L${plugindir} -lOgreMain@OGRE_LIB_SUFFIX@ @OGRE_ADDITIONAL_LIBS@
+Requires: 
+Libs: -L${libdir} -L${plugindir} -lOgreMain@OGRE_LIB_SUFFIX@
 Cflags: -I${includedir} -I${includedir}/OGRE @OGRE_CFLAGS@
diff -r 7c776867621e -r 2b38c1d04950 CMakeLists.txt
--- a/CMakeLists.txt	Fri Jun 20 00:39:55 2014 +0200
+++ b/CMakeLists.txt	Fri Aug 01 10:56:55 2014 +0200
@@ -514,7 +514,7 @@
 
 #Add android JNI binding
 if(ANDROID)
-  include(toolchain/AndroidJNI)
+#  include(toolchain/AndroidJNI)
 endif()
 
 # Setup command-line tools
diff -r 7c776867621e -r 2b38c1d04950 OgreMain/CMakeLists.txt
--- a/OgreMain/CMakeLists.txt	Fri Jun 20 00:39:55 2014 +0200
+++ b/OgreMain/CMakeLists.txt	Fri Aug 01 10:56:55 2014 +0200
@@ -146,7 +146,7 @@
 elseif(ANDROID)
   file(GLOB PLATFORM_HEADERS "${CMAKE_CURRENT_SOURCE_DIR}/include/Android/*.h")
   file(GLOB PLATFORM_SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/Android/*.cpp"
-      src/OgreSearchOps.cpp)
+      "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreSearchOps.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/src/Android/*.c")
 
   set(PLATFORM_HEADER_INSTALL "Android")
 elseif(FLASHCC)
